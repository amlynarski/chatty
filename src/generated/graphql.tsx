export type Maybe<T> = T | null;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  Date: any;
  Upload: any;
};

export enum CacheControlScope {
  Public = 'PUBLIC',
  Private = 'PRIVATE'
}

export type Conversation = {
   __typename?: 'Conversation';
  users: Array<User>;
  messages: Array<Message>;
  id: Scalars['ID'];
  lastMessageCreatedAt?: Maybe<Scalars['Date']>;
};


export type Message = {
   __typename?: 'Message';
  author: User;
  text: Scalars['String'];
  createdAt: Scalars['Date'];
  conversation: Conversation;
  readBy: Array<User>;
  id: Scalars['ID'];
};

export type Mutation = {
   __typename?: 'Mutation';
  sendMessage: Message;
  createConversation: Conversation;
  markMessageAsRead: Message;
  signin: Token;
};


export type MutationSendMessageArgs = {
  conversationId: Scalars['ID'];
  text: Scalars['String'];
};


export type MutationCreateConversationArgs = {
  userIds: Array<Scalars['ID']>;
};


export type MutationMarkMessageAsReadArgs = {
  messageId: Scalars['ID'];
};


export type MutationSigninArgs = {
  username: Scalars['String'];
  password: Scalars['String'];
};

export type Query = {
   __typename?: 'Query';
  me: User;
  messagesByConversationId: Array<Message>;
  conversations: Array<Conversation>;
  users: Array<User>;
};


export type QueryMessagesByConversationIdArgs = {
  id: Scalars['ID'];
};

export type Subscription = {
   __typename?: 'Subscription';
  messageAdded: Message;
};


export type SubscriptionMessageAddedArgs = {
  userId: Scalars['ID'];
};

export type Token = {
   __typename?: 'Token';
  jwt: Scalars['String'];
};


export type User = {
   __typename?: 'User';
  id: Scalars['ID'];
  username: Scalars['String'];
};

export type SigninMutationVariables = {
  username: Scalars['String'];
  password: Scalars['String'];
};


export type SigninMutation = (
  { __typename?: 'Mutation' }
  & { signin: (
    { __typename?: 'Token' }
    & Pick<Token, 'jwt'>
  ) }
);

export type MeQueryVariables = {};


export type MeQuery = (
  { __typename?: 'Query' }
  & { me: (
    { __typename?: 'User' }
    & Pick<User, 'id' | 'username'>
  ) }
);

export type ConversationsQueryVariables = {};


export type ConversationsQuery = (
  { __typename?: 'Query' }
  & { conversations: Array<(
    { __typename?: 'Conversation' }
    & Pick<Conversation, 'id' | 'lastMessageCreatedAt'>
    & { users: Array<(
      { __typename?: 'User' }
      & Pick<User, 'id' | 'username'>
    )>, messages: Array<(
      { __typename?: 'Message' }
      & Pick<Message, 'id'>
      & { author: (
        { __typename?: 'User' }
        & Pick<User, 'id'>
      ), readBy: Array<(
        { __typename?: 'User' }
        & Pick<User, 'id'>
      )> }
    )> }
  )> }
);

export type MessagesByConversationIdQueryVariables = {
  conversationId: Scalars['ID'];
};


export type MessagesByConversationIdQuery = (
  { __typename?: 'Query' }
  & { messagesByConversationId: Array<(
    { __typename?: 'Message' }
    & Pick<Message, 'id' | 'text' | 'createdAt'>
    & { author: (
      { __typename?: 'User' }
      & Pick<User, 'id' | 'username'>
    ), readBy: Array<(
      { __typename?: 'User' }
      & Pick<User, 'id' | 'username'>
    )> }
  )> }
);

export type SendMessageMutationVariables = {
  conversationId: Scalars['ID'];
  text: Scalars['String'];
};


export type SendMessageMutation = (
  { __typename?: 'Mutation' }
  & { sendMessage: (
    { __typename?: 'Message' }
    & Pick<Message, 'id' | 'text' | 'createdAt'>
    & { author: (
      { __typename?: 'User' }
      & Pick<User, 'id' | 'username'>
    ), readBy: Array<(
      { __typename?: 'User' }
      & Pick<User, 'id' | 'username'>
    )> }
  ) }
);

export type MessageAddedSubscriptionVariables = {
  userId: Scalars['ID'];
};


export type MessageAddedSubscription = (
  { __typename?: 'Subscription' }
  & { messageAdded: (
    { __typename?: 'Message' }
    & Pick<Message, 'id' | 'text' | 'createdAt'>
    & { author: (
      { __typename?: 'User' }
      & Pick<User, 'id' | 'username'>
    ), readBy: Array<(
      { __typename?: 'User' }
      & Pick<User, 'id' | 'username'>
    )>, conversation: (
      { __typename?: 'Conversation' }
      & Pick<Conversation, 'id' | 'lastMessageCreatedAt'>
      & { users: Array<(
        { __typename?: 'User' }
        & Pick<User, 'id' | 'username'>
      )> }
    ) }
  ) }
);

export type CreateConversationMutationVariables = {
  userIds: Array<Scalars['ID']>;
};


export type CreateConversationMutation = (
  { __typename?: 'Mutation' }
  & { createConversation: (
    { __typename?: 'Conversation' }
    & Pick<Conversation, 'id' | 'lastMessageCreatedAt'>
    & { users: Array<(
      { __typename?: 'User' }
      & Pick<User, 'id' | 'username'>
    )>, messages: Array<(
      { __typename?: 'Message' }
      & Pick<Message, 'text' | 'createdAt' | 'id'>
      & { author: (
        { __typename?: 'User' }
        & Pick<User, 'id' | 'username'>
      ), readBy: Array<(
        { __typename?: 'User' }
        & Pick<User, 'id' | 'username'>
      )> }
    )> }
  ) }
);

export type MarkMessageAsReadMutationVariables = {
  messageId: Scalars['ID'];
};


export type MarkMessageAsReadMutation = (
  { __typename?: 'Mutation' }
  & { markMessageAsRead: (
    { __typename?: 'Message' }
    & Pick<Message, 'id' | 'text' | 'createdAt'>
    & { conversation: (
      { __typename?: 'Conversation' }
      & Pick<Conversation, 'id'>
    ), author: (
      { __typename?: 'User' }
      & Pick<User, 'id' | 'username'>
    ), readBy: Array<(
      { __typename?: 'User' }
      & Pick<User, 'id' | 'username'>
    )> }
  ) }
);

export type UsersQueryVariables = {};


export type UsersQuery = (
  { __typename?: 'Query' }
  & { users: Array<(
    { __typename?: 'User' }
    & Pick<User, 'id' | 'username'>
  )> }
);
