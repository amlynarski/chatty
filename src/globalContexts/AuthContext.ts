import React from 'react';

export interface AuthContextType {
  token: string | null,
  login(username: string, password: string): Promise<void>,
  logout(): Promise<void>
}

const emptyFn = () => undefined;

export const defaultAuthContext = {
  token: null,
  login: emptyFn,
  logout: emptyFn
};

export const AuthContext = React.createContext<AuthContextType>(defaultAuthContext);
