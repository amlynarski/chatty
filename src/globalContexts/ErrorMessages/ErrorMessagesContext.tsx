import React from 'react';

export interface ErrorMessagesContextType {
  showError: (error: any) => void
}

export const defaultErrorMessagesContext = {
  showError: () => undefined
};

export const ErrorMessagesContext = React.createContext<ErrorMessagesContextType>(defaultErrorMessagesContext);