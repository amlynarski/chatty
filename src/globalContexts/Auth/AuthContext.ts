import React from 'react';
import { User } from '../../generated/graphql';

export interface AuthContextType {
  token: string | null,
  loading: boolean,
  me: User | null,
  login(username: string, password: string): Promise<string>,
  logout(): Promise<void>
}

const emptyFn = () => undefined;

export const defaultAuthContext = {
  token: null,
  loading: true,
  me: null,
  login: emptyFn,
  logout: emptyFn
};

export const AuthContext = React.createContext<AuthContextType>(defaultAuthContext);
