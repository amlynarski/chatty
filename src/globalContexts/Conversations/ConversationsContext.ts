import React from 'react';
import { Conversation, Message } from '../../generated/graphql';

export interface ConversationsContextType {
  conversations: Conversation[],
  loading: boolean,
  error: string | null,
  createConversation: (userIds: string[]) => Promise<Conversation>,
  markAsRead: (messageId: string) => Promise<Message>,
  setTotalUnread: (value: number) => void,
  totalUnread: number
}

const emptyFn = () => undefined;

export const defaultConversationsContext = {
  conversations: [],
  loading: false,
  error: null,
  createConversation: emptyFn,
  markAsRead: emptyFn,
  setTotalUnread: emptyFn,
  totalUnread: 0
};

export const ConversationsContext = React.createContext<ConversationsContextType>(defaultConversationsContext);
