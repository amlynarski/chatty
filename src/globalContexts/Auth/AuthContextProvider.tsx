import React, { PropsWithChildren, useEffect, useState } from 'react';
import * as SecureStore from 'expo-secure-store';
import { useMutation, useApolloClient } from '@apollo/react-hooks';
import { deleteItemAsync, getItemAsync } from 'expo-secure-store';

import { AuthContext } from './AuthContext';
import { ME, SIGNIN } from './schemas/auth.schema';
import { MeQuery, MeQueryVariables, SigninMutation, SigninMutationVariables, User } from '../../generated/graphql';

export const AuthContextProvider = (props: PropsWithChildren<{}>) => {
  const [token, setToken] = useState<string | null>(null);
  const [loading, setLoading] = useState<boolean>(true);
  const [me, setMe] = useState<User | null>(null);

  const apolloClient = useApolloClient();

  async function getMeUser(): Promise<User> {
    const response = await apolloClient.query<MeQuery, MeQueryVariables>({
      query: ME
    });
    return response.data.me;
  }

  useEffect(() => {
    apolloClient.cache.reset();
  }, []);

  useEffect(() => {
    async function checkIfTokenExist() {
      const jwt = await getItemAsync('token');
      if (jwt) {
        setToken(jwt);
        const meUser = await getMeUser();
        setMe(meUser);
      }
      setLoading(false);
    }

    checkIfTokenExist();
  }, []);

  useEffect(() => {
    async function clearData() {
      await apolloClient.cache.reset();
      setMe(null);
    }

    if (!token) {
      clearData()
    }
  }, [token]);

  const [signin] = useMutation<SigninMutation, SigninMutationVariables>(SIGNIN);

  async function login(username: string, password: string): Promise<string> {
    const response = await signin( {variables: {username, password}});
    await SecureStore.setItemAsync('token', response.data.signin.jwt);
    setToken(response.data.signin.jwt);
    const meUser = await getMeUser();
    setMe(meUser);
    return Promise.resolve(response.data.signin.jwt);
  }

  async function logout(): Promise<void> {
    await deleteItemAsync('token');
    setToken(null);
  }

  return(
    <AuthContext.Provider
      value={{
        login,
        logout,
        token,
        loading,
        me
      }}
    >
      {props.children}
    </AuthContext.Provider>
  )
};
