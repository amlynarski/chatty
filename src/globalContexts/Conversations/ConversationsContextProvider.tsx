import React, { PropsWithChildren, useContext, useEffect, useState } from 'react';
import { useApolloClient, useMutation, useQuery } from '@apollo/react-hooks';

import { ConversationsContext } from './ConversationsContext';
import { AuthContext, AuthContextType } from '../Auth/AuthContext';
import {
  Conversation,
  ConversationsQuery,
  ConversationsQueryVariables,
  CreateConversationMutation,
  CreateConversationMutationVariables,
  MarkMessageAsReadMutation, MarkMessageAsReadMutationVariables, Message,
  MessageAddedSubscription,
  MessageAddedSubscriptionVariables, MessagesByConversationIdQuery, MessagesByConversationIdQueryVariables
} from '../../generated/graphql';
import {
  CONVERSATIONS,
  CREATE_CONVERSATION, MARK_MESSAGE_AS_READ,
  MESSAGES,
  ON_MESSAGE_ADDED
} from '../../screens/conversations/conversations.schema';
import { NetworkStatusContext, NetworkStatusContextType } from '../NetworkStatus/NetworkStatusContext';
import { getUpdatedConversationsWithReadBy } from './utils';

export const ConversationsContextProvider = (props: PropsWithChildren<{}>) => {
  const { me } = useContext<AuthContextType>(AuthContext);
  const apolloClient = useApolloClient();
  const {connected} = useContext<NetworkStatusContextType>(NetworkStatusContext);
  const [newMessageData, setNewMessageData] = useState<MessageAddedSubscription | null>(null);
  const [totalUnread, setTotalUnread] = useState<number>(0);

  const { loading, error, data: conversationsData, refetch } = useQuery<ConversationsQuery, ConversationsQueryVariables>(CONVERSATIONS);

  useEffect(() => {
    let subscription;
    function subscribe() {
      const observer = apolloClient.subscribe<MessageAddedSubscription, MessageAddedSubscriptionVariables>({
        query: ON_MESSAGE_ADDED,
        variables: {
          userId: me.id
        }
      });

      subscription = observer.subscribe(({data}) => {
        setNewMessageData(data);
      })
    }

    if (me && connected) {
      subscribe();
      refetch();
    }

    return () => {
      if (subscription) {
        subscription.unsubscribe();
      }
    }

  }, [me, connected]);

  useEffect(() => {
    if (newMessageData) {
      const updatedConversation = conversationsData.conversations.find(conv => conv.id === newMessageData.messageAdded.conversation.id);
      if (!updatedConversation) {
        apolloClient.writeQuery({
          query: CONVERSATIONS,
          data: {
            conversations: [...conversationsData.conversations, {
              ...newMessageData.messageAdded.conversation,
              messages: [newMessageData.messageAdded]
            }]
          }
        })
      } else {
        try {
          const messagesData = apolloClient.readQuery<MessagesByConversationIdQuery, MessagesByConversationIdQueryVariables>(
            { query: MESSAGES, variables: {conversationId: updatedConversation.id} }
          );
          apolloClient.writeQuery({
            query: MESSAGES,
            variables: {conversationId: updatedConversation.id},
            data: {
              messagesByConversationId: [...messagesData.messagesByConversationId, newMessageData.messageAdded]
            }
          });
        } catch {
          // no data in cache
        }

        apolloClient.writeQuery({
          query: CONVERSATIONS,
          data: {
            conversations: conversationsData.conversations.map(conv => {
              if (conv.id === updatedConversation.id) {
                return ({
                  ...conv,
                  messages: [
                    ...conv.messages,
                    newMessageData.messageAdded
                  ],
                  lastMessageCreatedAt: newMessageData.messageAdded.createdAt
                })
              } else {
                return conv;
              }
            })
          }
        })
      }
    }
  }, [newMessageData]);

  const [_createConversation, {loading: createConversationLoading}] = useMutation<
    CreateConversationMutation,
    CreateConversationMutationVariables>(
    CREATE_CONVERSATION,
    {
      update(cache, {data: {createConversation}}) {
        const {conversations} = cache.readQuery<ConversationsQuery, ConversationsQueryVariables>({query: CONVERSATIONS});
        cache.writeQuery<ConversationsQuery, ConversationsQueryVariables>({
          query: CONVERSATIONS,
          data: {
            conversations: conversations.concat([createConversation])
          }
        })
      }
    }
  );

  const [_markMessageAsRead] = useMutation<
    MarkMessageAsReadMutation,
    MarkMessageAsReadMutationVariables>(
    MARK_MESSAGE_AS_READ,
    {
      update(cache, {data: {markMessageAsRead}}) {
        const {conversations} = cache.readQuery<ConversationsQuery, ConversationsQueryVariables>({query: CONVERSATIONS});
        cache.writeQuery<ConversationsQuery, ConversationsQueryVariables>({
          query: CONVERSATIONS,
          data: {
            conversations: getUpdatedConversationsWithReadBy(conversations as Conversation[], markMessageAsRead as Message)
          }
        })
      }
    }
  );

  async function createConversation (userIds: string[]): Promise<Conversation> {
    const createdResponse = await _createConversation({variables: {userIds}});
    return createdResponse.data.createConversation as Conversation;
  }

  async function markAsRead (messageId: string): Promise<Message> {
    const response = await _markMessageAsRead({variables: {messageId}});
    return response.data.markMessageAsRead as Message;
  }

  return (
    <ConversationsContext.Provider
      value={{
        conversations: conversationsData ? conversationsData.conversations as Conversation[] : [],
        createConversation,
        loading,
        error: error ? error.message : null,
        markAsRead,
        totalUnread,
        setTotalUnread
      }}
    >
      {props.children}
    </ConversationsContext.Provider>
  )
};
