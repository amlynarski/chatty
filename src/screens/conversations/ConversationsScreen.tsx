import { View } from 'react-native';
import React, { useContext, useEffect } from 'react';
import { List, Text } from 'react-native-paper';

import {
  Conversation
} from '../../generated/graphql';
import { AuthContext, AuthContextType } from '../../globalContexts/Auth/AuthContext';
import { ConversationsContext, ConversationsContextType } from '../../globalContexts/Conversations/ConversationsContext';
import getFormattedTime from '../../utils/getFormattedTime';
import { Badge } from '../../components/Badge';
import { ErrorMessagesContext, ErrorMessagesContextType } from '../../globalContexts/ErrorMessages/ErrorMessagesContext';
import { countNotReadMessages } from './utils';

function leftListElement(unreadCount: number) {
  return (
    <>
      <List.Icon icon="email" />
      {
        unreadCount > 0 && <Badge value={unreadCount}/>
      }
    </>
  )
}

export default function ConversationsScreen({navigation}) {
  const { me } = useContext<AuthContextType>(AuthContext);
  const { loading, error, conversations, setTotalUnread } = useContext<ConversationsContextType>(ConversationsContext);
  const {showError} = useContext<ErrorMessagesContextType>(ErrorMessagesContext);

  useEffect(() => {
    if (error) {
      showError(error);
    }
  }, [error]);

  useEffect(() => {
    if (conversations) {
      const totalUnread = conversations.reduce((acc, curr) => acc + countNotReadMessages(curr, me), 0);
      setTotalUnread(totalUnread);
    }
  }, [conversations]);

  function conversationUserNames(conversation: Conversation): String {
    const usersWithoutMe = conversation.users.filter(user => user.id !== me.id);
    return usersWithoutMe.map(user => user.username).join(', ');
  }

  function openConversation(conversationId: string) {
    navigation.navigate('Message', {conversationId});
  }

  function conversationItem(conversation: Conversation) {
    return (
      <List.Item
        key={conversation.id}
        onPress={() => openConversation(conversation.id)}
        title={conversationUserNames(conversation)}
        left={() => leftListElement(countNotReadMessages(conversation, me))}
        description={getFormattedTime(conversation.lastMessageCreatedAt)}
        descriptionStyle={{'fontSize': 10}}
      />
    )
  }

  if (!me) {
    return null;
  }

  return (
    <View>
      <List.Section>
        {
          conversations
            .sort((a, b) => b.lastMessageCreatedAt - a.lastMessageCreatedAt)
            .map(conversation => conversationItem(conversation as Conversation))}
      </List.Section>
      {
        !loading && conversations && conversations.length === 0 &&
        <View style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
          <Text>
            You don't have any messages yet.
          </Text>
        </View>
      }
    </View>
  );
}
