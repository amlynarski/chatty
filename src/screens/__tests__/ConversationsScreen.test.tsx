import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import { List } from 'react-native-paper';

import ConversationsScreen from '../conversations/ConversationsScreen';
import { AuthContext } from '../../globalContexts/Auth/AuthContext';
import { ConversationsContext } from '../../globalContexts/Conversations/ConversationsContext';
import { ErrorMessagesContext } from '../../globalContexts/ErrorMessages/ErrorMessagesContext';
import { Conversation } from '../../generated/graphql';

const navigateMock = jest.fn();

const me = {
  username: 'testuser',
  id: 'ID_USER_TEST_USER'
};

const users = [
  {
    username: 'testuser',
    id: 'ID_USER_TEST_USER'
  },
  {
    username: 'adam',
    id: 'ID_USER_ADAM'
  }
];

const conversations = [
  {
    id: 'conv1',
    users: [users[0], users[1]],
    messages: [
      {
        author: users[0],
        text: 'lorem ipsum ',
        createdAt: new Date(1585129870344),
        readBy: [],
        id: "rand1"
      },
      {
        author: users[0],
        text: 'Other text',
        createdAt: new Date(1585517339938),
        readBy: [],
        id: "rand2"
      }
    ],
    lastMessageCreatedAt: new Date(1585517339938)
  },
  {
    id: 'conv2',
    users: [users[0], users[1]],
    lastMessageCreatedAt: new Date(1585129878644),
    messages: [
      {
        author: users[0],
        text: 'test 1',
        createdAt: new Date(1585129870344),
        readBy: [users[0]],
        id: "rand3"
      },
      {
        author: users[1],
        text: 'test 2',
        createdAt: new Date(1585129878344),
        readBy: [users[0]],
        id: "rand4"
      },
      {
        author: users[1],
        text: 'test 3',
        createdAt: new Date(1585129878644),
        readBy: [],
        id: "rand5"
      }
    ]
  }
];

const ConversationsWithProviders = () => (
  <AuthContext.Provider value={{me, token: '', loading: false, login: jest.fn(), logout: jest.fn()}}>
    <ConversationsContext.Provider value={{
      conversations: conversations as Conversation[],
      loading: false,
      error: null,
      createConversation: jest.fn(),
      markAsRead: jest.fn(),
      setTotalUnread: jest.fn(),
      totalUnread: 0
    }}>
      <ErrorMessagesContext.Provider value={{showError: jest.fn()}}>
        <ConversationsScreen navigation={{navigate: navigateMock}} />
      </ErrorMessagesContext.Provider>
    </ConversationsContext.Provider>
  </AuthContext.Provider>
);

describe('ConversationsScreen', () => {
  it('renders correctly', () => {
    const conversationsScreen = renderer.create(<ConversationsWithProviders />).toJSON();
    expect(conversationsScreen).toMatchSnapshot()
  });

  it('should display two conversations', () => {
    const instance = renderer.create(<ConversationsWithProviders />);
    const listItems = instance.root.findAllByType(List.Item);

    expect(listItems.length).toBe(2);
  });

  it('should call navigate function on click', () => {
    const instance = renderer.create(<ConversationsWithProviders />);
    const listItems = instance.root.findAllByType(List.Item);

    listItems[0].props.onPress();

    expect(navigateMock).toHaveBeenCalledWith("Message", {"conversationId": "conv1"});
  });
});
