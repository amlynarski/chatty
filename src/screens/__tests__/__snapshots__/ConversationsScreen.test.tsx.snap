// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationsScreen renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "marginVertical": 8,
        },
        undefined,
      ]
    }
    theme={
      Object {
        "animation": Object {
          "scale": 1,
        },
        "colors": Object {
          "accent": "#03dac4",
          "backdrop": "rgba(0, 0, 0, 0.5)",
          "background": "#f6f6f6",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "error": "#B00020",
          "notification": "#f50057",
          "onBackground": "#000000",
          "onSurface": "#000000",
          "placeholder": "rgba(0, 0, 0, 0.54)",
          "primary": "#6200ee",
          "surface": "#ffffff",
          "text": "#000000",
        },
        "dark": false,
        "fonts": Object {
          "light": Object {
            "fontFamily": "System",
            "fontWeight": "300",
          },
          "medium": Object {
            "fontFamily": "System",
            "fontWeight": "500",
          },
          "regular": Object {
            "fontFamily": "System",
            "fontWeight": "400",
          },
          "thin": Object {
            "fontFamily": "System",
            "fontWeight": "100",
          },
        },
        "roundness": 4,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Array [
            Object {
              "padding": 8,
            },
            undefined,
          ],
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "margin": 8,
                "width": 40,
              },
              undefined,
            ]
          }
        >
          <Text />
        </View>
        <View
          style={
            Array [
              Object {
                "marginVertical": 6,
                "paddingLeft": 8,
              },
              Object {
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "fontSize": 16,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                  },
                  undefined,
                ],
              ]
            }
          >
            adam
          </Text>
          <Text
            numberOfLines={2}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "fontSize": 14,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                  },
                  Object {
                    "fontSize": 10,
                  },
                ],
              ]
            }
          >
            29.3.2020 23:28
          </Text>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Array [
            Object {
              "padding": 8,
            },
            undefined,
          ],
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "margin": 8,
                "width": 40,
              },
              undefined,
            ]
          }
        >
          <Text />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#f00",
              "borderRadius": 20,
              "left": 30,
              "opacity": 0.7,
              "paddingBottom": 1,
              "paddingLeft": 5,
              "paddingRight": 5,
              "paddingTop": 1,
              "position": "absolute",
              "top": 10,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Object {
                  "color": "#fff",
                  "fontSize": 8,
                },
              ]
            }
          >
            1
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "marginVertical": 6,
                "paddingLeft": 8,
              },
              Object {
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "fontSize": 16,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                  },
                  undefined,
                ],
              ]
            }
          >
            adam
          </Text>
          <Text
            numberOfLines={2}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "fontSize": 14,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                  },
                  Object {
                    "fontSize": 10,
                  },
                ],
              ]
            }
          >
            25.3.2020 10:51
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
